src/object.d

```diff
@@ -1431,6 +1431,132 @@ class TypeInfo_Struct : TypeInfo
     assert(!typeid(S).equals(&s, &s));
 }
 
+class RTTypeid(T) : TypeInfo
+if (is(T == struct))
```

Comment:
The current compiler implemtation in https://github.com/dlang/dmd/pull/11459 does not allow selectively replacing TypeInfo, it's either all or none.

Reply:
That's right. I thought a good start would be to have RTTypeid duplicate what the compiler does, get everything working, and then start improving what it does. I.e. start by implementing instantiations of the TypeInfoXXXX classes.